src/backend/dwarf.c

```diff
@@ -100,7 +100,7 @@ void dwarf_addrel(int seg, targ_size_t offset, int targseg, targ_size_t val = 0)
 void dwarf_addrel64(int seg, targ_size_t offset, int targseg, targ_size_t val)
 {
 #if ELFOBJ
-    ElfObj::addrel(seg, offset, R_X86_64_64, MAP_SEG2SYMIDX(targseg), val);
+    ElfObj::addrel(seg, offset, I64 ? R_X86_64_64 : R_386_32, MAP_SEG2SYMIDX(targseg), val);
```

Comment:
I know, it used to be separated in dwarf_appreladdr. But after your refactor (https://github.com/D-Programming-Language/dmd/commit/4c759b924cd6221e5892e183775a509e05923e7d), now when you call dwarf_appreladdr() in 32 bit mode, it calls dwarf_addrel64. The value of R_X86_64_64 is the same as R_386_32, so it works.

You can argue that I should not call dwarf_appreladdr in 32 bit mode, but I think such 32b/64b agnostic function is useful.